lib/src/rules/missing_whitespace_between_adjacent_strings.dart

```diff
@@ -119,6 +94,61 @@ class _Visitor extends RecursiveAstVisitor<void> {
   }
 }
 
+extension on StringLiteral {
+  /// Returns whether this ends with whitespace, where any
+  /// [InterpolationExpression] counts as whitespace.
+  bool get endsWithWhitespace {
+    if (this is SimpleStringLiteral) {
+      return (this as SimpleStringLiteral).value.endsWithWhitespace;
+    } else if (this is StringInterpolation) {
+      var last = (this as StringInterpolation).elements.last;
+      if (last is InterpolationExpression) {
```

Comment:
This condition won't ever be true. The first and last elements in a `StringInterpolation` are always an `InterpolationString` (see `StringInterpolation.firstString` and `StringInterpolation.lastString`). In the case of a string interpolation such as `'$a'`, both the first and last elements will be empty strings. As a result, I think the code needs to consider an empty `InterpolationString` to be the same as whitespace.

Note that there are no tests for the case where an interpolation expression is either first or last but the other end of the string contains whitespace. For example: `'a $b' 'c'` or `'a' '$b c'`.

Reply:
Ah ha good catch. Much simpler now.